demo/ZeroToHeroUsingTheBasis.md

```diff
@@ -0,0 +1,357 @@
+
+## Questions:
+* I dont seem to get this event: Alice's Agent Verifies that Faber has Received the Message
+That the message has been received by the agent, might this be a multitenant error? It is only one webhook message that it has been received on the recipient side.
+
+# Issuance
+As the demo says it does for you, setting up all for issuance: https://github.com/hyperledger/aries-cloudagent-python/blob/main/demo/AriesOpenAPIDemo.md#preparing-to-issue-a-credential we will manually do here so we understand what is really behind this.
+
+Lets say we are using this ledger: http://test.bcovrin.vonx.io/
+According to your configuration from earlier, we are.
+
+**PS:** No webhooks on the initial setup of schema and credential definition, so there might have to be a controller setup for controlling the creation of these schemas so that you can have some stats on it. Always possible to query all wallets, but that is not good. So at some of these controller endpoints, it will be smart to have some db storing, audit logging to make sure that we know what is happening.
+
+## Creating the public did.
+Then we first need to create a DID, we can also use the one created for your connection, but we want to keep things seperated and not nessecarily correlateable.
+PS: Where the demo forces ngrok on you, so that you create an endpoint that is accessible by the web. But we dont need to consider this at the moment, because we are doing localhost, and endpoints can be rotated later when needed.
+
+Create a did with `POST /wallet/did/create` [Localhost link to API doc](http://localhost:5000/api/doc#/wallet/post_wallet_did_create)
+
+Go to http://test.bcovrin.vonx.io/, input the info from the response above.
+Then go and assign the did you have created, and put on the ledger, as public: `POST /wallet/did/public` [Localhost link to API doc](http://localhost:5000/api/doc#/wallet/post_wallet_did_public)
+
+**A public did we have!**
+This can be done programatically and all, and is very dependent on the ledger you will be using.
+
+All the next steps has to be done once, for each credential schema type. But once done for one credential type, you can issue as many as you want on that schema.
+
+## Creating the schema
+A schema is needed to make sure you know the attributes of a credential. This defines the names of the attributes and the schema, and also the version of the schema. Which is used in the credential definition
+[Localhost link to API doc](http://localhost:5000/api/doc#/schema/post_schemas)
+
+### Questions
+Are there any work going on in defining types in the schema or what is the 2 cents there?
+
+## Creating the credential definition 
+This is where you define if you want revocation or not. It also where you define the definition of the credential, based on a schema. [Localhost link to API doc](http://localhost:5000/api/doc#/credential-definition/post_credential_definitions)
+
+### Questions
+Why is this needed, except for revocation?
```

Comment:
In the AnonCreds verifiable credential mechanism -- the cred def links the issuer of the credential, the schema being used and contains a public key (and the issuer has the corresponding private key) for each attribute.  As noted it also provides an association with the revocation registries.

Reply:
https://github.com/hyperledger/aries-cloudagent-python/pull/1004/commits/26dfcb6a1157b74513fcb348303d8c61545188f2